examples/single_active_consumer/single_active_consumer_super_stream.rs

```diff
@@ -0,0 +1,79 @@
+use futures::StreamExt;
+use rabbitmq_stream_client::error::StreamCreateError;
+use rabbitmq_stream_client::types::{
+    ByteCapacity, OffsetSpecification, ResponseCode, SuperStreamConsumer,
+};
+use std::collections::HashMap;
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    use rabbitmq_stream_client::Environment;
+    let environment = Environment::builder().build().await?;
+    let message_count = 1000000;
+    let super_stream = "hello-rust-super-stream";
+
+    let create_response = environment
+        .stream_creator()
+        .max_length(ByteCapacity::GB(5))
+        .create_super_stream(super_stream, 3, None)
+        .await;
+
+    if let Err(e) = create_response {
+        if let StreamCreateError::Create { stream, status } = e {
+            match status {
+                // we can ignore this error because the stream already exists
+                ResponseCode::StreamAlreadyExists => {}
+                err => {
+                    println!("Error creating stream: {:?} {:?}", stream, err);
+                }
+            }
+        }
+    }
+    println!(
+        "Super stream consumer example, consuming messages from the super stream {}",
+        super_stream
+    );
+
+    let mut properties = HashMap::new();
+
+    properties.insert("single-active-consumer".to_string(), "true".to_string());
```

Comment:
All these properties should be hidden from the user:
```
properties.insert("single-active-consumer".to_string(), "true".to_string());
properties.insert("name".to_string(), "consumer-group-1".to_string());
properties.insert("super-stream".to_string(), "hello-rust-super-stream".to_string());
```


We should add a flag, like:
https://github.com/rabbitmq/rabbitmq-stream-dotnet-client/blob/main/docs/SuperStream/SuperStreamConsumer.cs#L50

About the name, we should use the consumer name:
```
properties.insert("name".to_string(), "consumer-group-1".to_string());
```

Like we do in the other clients